python-main.js

```diff
@@ -52,6 +52,55 @@ function pythonEditor(id) {
     ACE.setFontSize(editor.initialFontSize);
     editor.ACE = ACE;
 
+    // Add custom snippets
+    var snippetManager = ace.require("ace/snippets").snippetManager;
+    var microbitSnippets = [
+        '# Module Docstring',
+        'snippet docs - create a carlos to describe your code',
```

Comment:
Why does this say "create a carlos to describe your code"?

Reply:
😆 so glad you caught that! 
At one point I wasn't seeing changes on the displayed info, so I edited that to see if I could easily catch the different text. Might be a bit narcissistic, but since my name is not usually on the source code it makes it an easy search keyword to find any temporary code used during debugging.

This should be the original array:

```js
        snippetManager.parseSnippetFile(['# Module Docstring',
            'snippet docs - create a comment to describe your code',
            '\t\'\'\'',
            '\t${1:# TODO: write some helpful comments here...}',
            '\t\'\'\'',
            'snippet wh - while some condition is True, keep looping over some code',
            '\twhile ${1:condition}:',
            '\t\t${2:# TODO: write code...}',
            'snippet with - do stuff with something assigned to a name',
            '\twith ${1:something} as ${2:name}:',
            '\t${3:# TODO: write code...}',
            '# New Class',
            'snippet cl - create a new class that defines the behaviour of a new type of object',
            '\tclass ${1:ClassName}(${2:object}):',
            '\t\t\"\"\"${3:docstring for $1}\"\"\"',
            '\t\tdef __init__(self, ${4:arg}):',
            '\t\t\t${5:super($1, self).__init__()}',
            '\t\t\tself.$4 = $4',
            '\t\t\t${6}',
            '# New Function',
            'snippet def - define a named function that takes some arguments and optionally add a description',
            '\tdef ${1:name}(${2:arguments}):',
            '\t\t\"\"\"${3:description for $1}\"\"\"',
            '\t\t${4:# TODO: write code...}',
            '# Ifs',
            'snippet if - if some condition is True, do something',
            '\tif ${1:condition}:',
            '\t\t${2:# TODO: write code...}',
            'snippet ei - else if some other condition is True, do something',
            '\telif ${1:condition}:',
            '\t\t${2:# TODO: write code...}',
            'snippet el - else do some other thing',
            '\telse:',
            '\t\t${1:# TODO: write code...}',
            '# For',
            'snippet for - for each item in a collection of items do something with each item',
            '\tfor ${1:item} in ${2:items}:',
            '\t\t${3:# TODO: write code...}',
            'snippet try - try doing something and handle exceptions (errors)',
            '\ttry:',
            '\t\t${1:# TODO: write code...}',
            '\texcept ${2:Exception}, ${3:e}:',
            '\t\t${4:raise $3}',
            ''].join('\n'));
```

Reply:
Oddly, though, I don't see "Carlos" anywhere when I cloned this locally.

Reply:
I've updated these snippets to the ones above. I think that's everything?

Reply:
Yes, looks like this might be ready for beta